packages/typespec-test/test/openai_modular/generated/typespec-ts/src/api/operations.ts

```diff
@@ -268,19 +272,7 @@ export async function _getChatCompletionsDeserialize(
     id: result.body["id"],
     created: new Date(result.body["created"]),
     choices: (result.body["choices"] ?? []).map((p) => ({
-      message: !p.message
-        ? undefined
-        : {
-            role: p.message?.["role"],
-            content: p.message?.["content"],
-            name: p.message?.["name"],
-            functionCall: !p.message?.function_call
-              ? undefined
-              : {
-                  name: p.message?.function_call?.["name"],
-                  arguments: p.message?.function_call?.["arguments"],
-                },
-          },
+      message: !p.message ? undefined : (p.message as any),
```

Comment:
This is buggy because `functionCall` will be replaced by `function_call`.

Reply:
Ignore my comment, I will need to use the custom deserializer you provided here anyway.